src/clarity-angular/datagrid/datagrid.html

```diff
@@ -5,59 +5,62 @@
+                        <div class="datagrid-column datagrid-expandable-caret datagrid-fixed-column"
+                             *ngIf="expandableRows.hasExpandableRow">
+                            <div class="datagrid-column-separator"></div>
+                        </div>
+                        <ng-content select="clr-dg-column"></ng-content>
+                    </div>
                 </div>
-                <!-- header for single row action; only display if we have at least one actionable row in datagrid -->
-                <div class="datagrid-column datagrid-row-actions datagrid-fixed-column"
-                     *ngIf="rowActionService.hasActionableRow">
-                    <div class="datagrid-column-separator"></div>
-                </div>
-                <!-- header for carets; only display if we have at least one expandable row in datagrid -->
-                <div class="datagrid-column datagrid-expandable-caret datagrid-fixed-column"
-                     *ngIf="expandableRows.hasExpandableRow">
-                    <div class="datagrid-column-separator"></div>
+
+                <div clrDgBody class="datagrid-body">
+                    <ng-template *ngIf="iterator"
+                                 ngFor [ngForOf]="items.displayed" [ngForTrackBy]="items.trackBy"
+                                 [ngForTemplate]="iterator.template"></ng-template>
+                    <ng-content *ngIf="!iterator"></ng-content>
+
+                    <!-- Custom placeholder overrides the default empty one -->
+                    <ng-content select="clr-dg-placeholder"></ng-content>
+                    <clr-dg-placeholder *ngIf="!placeholder"></clr-dg-placeholder>
                 </div>
-                <ng-content select="clr-dg-column"></ng-content>
             </div>
-        </div>
-
-        <div clrDgBody class="datagrid-body">
-            <ng-template *ngIf="iterator"
-                      ngFor [ngForOf]="items.displayed" [ngForTrackBy]="items.trackBy"
-                      [ngForTemplate]="iterator.template"></ng-template>
-            <ng-content *ngIf="!iterator"></ng-content>
 
-            <!-- Custom placeholder overrides the default empty one -->
-            <ng-content select="clr-dg-placeholder"></ng-content>
-            <clr-dg-placeholder *ngIf="!placeholder"></clr-dg-placeholder>
+            <!--
+                This is not inside the table because there is no good way of having a single column span
+                everything when using custom elements with display:table-cell.
+            -->
+            <ng-content select="clr-dg-footer"></ng-content>
+    </div>
+    <div class="datagrid-spinner-wrapper" *ngIf="loading">
+        <!--we are using the offsetHeight here to set the height of spinner which is brittle;
+            re-evaluate later to see if there is a better fix for this-->
+        <div class="datagrid-spinner">
```

Comment:
With the new CSS you have, you don't need the spinner wrapper anymore. Just:
```html
<div class="datagrid-spinner" *ngIf="loading">
    <div class="spinner">Loading...</div>
</div>
```
and moving the `.datagrid-spinner-wrapper` styles to `.datagrid-spinner` itself should work.